sds/src/secondary_validation/luxembourg_individual_nin_checksum.rs

```diff
@@ -0,0 +1,90 @@
+use crate::secondary_validation::{LuhnChecksum, Validator, VerhoeffChecksum};
+
+pub struct LuxembourgIndividualNINChecksum;
+
+const LUXEMBOURG_INDIVIDUALS_NIN_LENGTH: usize = 13;
+
+impl Validator for LuxembourgIndividualNINChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        /*
+         * Luxembourg uses a 13-digit identification code,
+         * consisting of the birth date formatted as YYYYMMDD followed by a number XXX
+         * ensuring persons born on the same date have a unique national ID,
+         * then a first check on YYYYMMDDXXX using the Luhn10 algorithm,
+         * and finally a check on YYYYMMDDXXX using the Verhoeff algorithm.
+         */
+        if regex_match.len() != LUXEMBOURG_INDIVIDUALS_NIN_LENGTH {
+            return false;
+        }
+
+        let verhoeff_digit = regex_match.chars().last().unwrap();
+        let luhn_digit = regex_match.chars().nth(regex_match.len() - 2).unwrap();
+        let nin_without_checksum = &regex_match[..regex_match.len() - 2];
```

Comment:
Direct string indexing is dangerous and can panic since you haven't verified `regex_match.len() - 2` is a valid char index yet (it could be the middle of a multi-byte character). Either use the `.chars()` iterator, or verify you are working with only ASCII chars first